repo/fsrepo/migrations/fetch.go

```diff
@@ -0,0 +1,269 @@
+package migrations
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"strings"
+)
+
+const (
+	envIpfsDistPath = "IPFS_DIST_PATH"
+
+	// Distribution
+	gatewayURL = "https://ipfs.io"
+	ipfsDist   = "/ipns/dist.ipfs.io"
```

Comment:
This is a security issue and should be changed to use an IPFS path.
```go
ipfsDist = "/ipfs/Qme8pJhBidEUXRdpcWLGR2fkG5kdwVnaMh3kabjfP8zz7Y"
```
The downside to this is that nothing newer than what is in that distribution can be fetched.  In the case of `ipfs-update` that means either downloading the latest `ipfs-update`, or specifying the distribution to fetch from (using `IPFS_DIST_PATH` environ var or `-distpath` option).

Reply:
I think for now we should keep existing behavior:
- `ipfs-update` pulls from the DNSLink record
- `go-ipfs` repo migrations pull from the version of dist bundled with the binary

I'd like to have an option for automatic updating of go-ipfs, but that's a separate issue + PR.

The short reasoning for why we're using different semantics for `ipfs-update` vs `go-ipfs` migrations is basically that go-ipfs migrations happen automatically whereas ipfs-update is something a user manually runs where the alternative is that they'd likely go to GitHub or dist.ipfs.io anyway.

Reply:
OK, I will have go-ipfs use the ipfs path and ipfs-update use a DNSLink record. What you prefer the default behavior of this library to be, ipfs or dnslink?

Reply:
Done.